Makefile

```diff
@@ -1,38 +1,25 @@
-CC=clang
-CXX=clang
-LD=ld.lld
-RUST=rustc
-
-CFLAGS=$(shell llvm-config --cflags) -emit-llvm -Iinclude
-CXXFLAGS=$(shell llvm-config --cxxflags) -emit-llvm
-LDFLAGS=$(shell llvm-config --ldflags)
-RUSTFLAGS=--crate-type=lib --emit=llvm-bc
-
-LDLIBS=stdc++
-OBJECTS=$(addprefix build/, $(addsuffix .ll, hello.c hello.cpp hello.rs main.c))
-
-.PHONY: all
-all: link
-
-.PHONY: link
-link: build
-	$(LD) $(LDFLAGS) $(addprefix -l, $(LDLIBS)) -o hello $(OBJECTS)
-
-.PHONY: build
-build: build/ $(OBJECTS)
-
-.PHONY: clean
-clean:
-	$(RM) $(OBJECTS) hello
-
-build/:
+.PHONY:
+run: build/hello
+	./build/hello
+
+build/hello: build/librust_hello.a build/hello_cpp.a
+	$(CC) src/main.c src/hello.c build/hello_cpp.a build/librust_hello.a -o build/hello -lstdc++
+build/librust_hello.a: src/rust-hello/*
+	sh -c "cd src/rust-hello &&\
+		cargo build --release &&\
+		cp target/release/librust_hello.a ../../build/librust_hello.a &&\
+		cd ../.."
```

Comment:
I don't like using sh in a Makefile

Reply:
I don't either, but Cargo requires you be under the same directory as the crate you're working with.